### DISCUSSION TITLE
Building MikTeX for MacOS

### DISCUSSION THREAD

**Discussion by memcat2** [MiKTeX/miktex#1404] [Q&A]
- [x] I have read the [MiKTeX Contributing Guidelines](https://github.com/MiKTeX/miktex-packaging/blob/master/CONTRIBUTING.md)

- I am a complete newbie to building MikTeX, so therefore the reported issue may be trivial to resolve.
- I realised that the MikTeX package for MacOS on miktex.org is older than the one for Windows.
- So I've tried to build MikTeX for MacOS according to this [link](https://miktex.org/howto/build-mac).
- I downloaded the miktex-23.10 source and I am running macOS 14.0.
- When running the build script, I get the following output, and I don't have any clue how to fix the issue:

% cmake ../miktex-23.10       
-- The C compiler identification is AppleClang 14.0.3.14030022
-- The CXX compiler identification is AppleClang 14.0.3.14030022
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/bin/git (found version "2.39.2 (Apple Git-143)") 
-- Found BISON: /usr/bin/bison (found version "2.3") 
-- Found FLEX: /usr/bin/flex (found version "2.6.4") 
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
CMake Error at CMakeLists.txt:1110 (find_package):
  By not providing "FindQt5.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Qt5", but
  CMake did not find one.

  Could not find a package configuration file provided by "Qt5" with any of
  the following names:

    Qt5Config.cmake
    qt5-config.cmake

  Add the installation prefix of "Qt5" to CMAKE_PREFIX_PATH or set "Qt5_DIR"
  to a directory containing one of the above files.  If "Qt5" provides a
  separate development package or SDK, be sure it has been installed.

-- Configuring incomplete, errors occurred!

**Comment by klefo0** [comment]
You can read on this link
https://stackoverflow.com/questions/40081327/macos-sierra-cmake-failed-with-qt5#comment87998134_40081874

**Comment by memcat2** [comment]
> You can read on this link https://stackoverflow.com/questions/40081327/macos-sierra-cmake-failed-with-qt5#comment87998134_40081874

Hi klefo0 - I looked at the link, but as a complete newbie, I am lost. I just followed the build instructions [here](https://miktex.org/howto/build-mac).

It seems as if there is something missing in the build instructions, but I can't figure out what.

**Comment by klefo0** [comment]
Hi @memcat2, you can see CMake  throw error said
> CMake Error at CMakeLists.txt:1110 (find_package):
>By not providing "FindQt5.cmake" in CMAKE_MODULE_PATH this project has
>asked CMake to find a package configuration file provided by "Qt5", but
>CMake did not find one.

In short, CMake not found Qt5 library on your machine.
So, at the [link](https://stackoverflow.com/questions/40081327/macos-sierra-cmake-failed-with-qt5#comment87998134_40081874) i gave you, it shows a way to help you install Qt5.

**Comment by memcat2** [comment]
> Hi @memcat2, you can see CMake throw error said
> 
> > CMake Error at CMakeLists.txt:1110 (find_package):
> > By not providing "FindQt5.cmake" in CMAKE_MODULE_PATH this project has
> > asked CMake to find a package configuration file provided by "Qt5", but
> > CMake did not find one.
> 
> In short, CMake not found Qt5 library on your machine. So, at the [link](https://stackoverflow.com/questions/40081327/macos-sierra-cmake-failed-with-qt5#comment87998134_40081874) i gave you, it shows a way to help you install Qt5.

Hi - thanks. I am not sure, whether Qt5 is not installed: I followed the [installation instructions](https://miktex.org/howto/build-mac) on the MikTeX page and it contains the command "brew install qt". When I enter "brew --prefix qt5", the output is "/opt/homebrew/opt/qt@5", which seems to be right (the folder exists & contains a substantial amount of files). Or is there anything wrong with this?

**Comment by klefo0** [comment]
@memcat2 the ```brew install qt``` would install newest QT version (is QT 6) so you need to input this ```brew install qt@5.11```

**Comment by Behinder** [comment]
Compiling Miktex with QT6 won't work and it does not have qtscript. Building for arm64 wiith qt511 also not work

**Comment by edocevoli** [comment]
Since this is not a bug per se, I will move this to discussions.

**Reply by Behinder** [reply]
If building fails then definitely it's a bug
